include/kernel.h

```diff
@@ -646,6 +646,11 @@ struct k_thread {
 	/** resource pool */
 	struct k_mem_pool *resource_pool;
 
+#if defined(CONFIG_THREAD_LOCAL_STORAGE)
+	/* Pointer to arch-specific TLS area */
+	uintptr_t tls;
```

Comment:
Why not just 'void *' ?

Reply:
We only need to store the address value, and does not need this to act as a pointer. It's a simple way of indicating to developers this is not to use it as straight pointer.

Reply:
That's a nice convention.